graphql_compiler/compiler/compiler_frontend.py

```diff
@@ -718,6 +718,21 @@ def _compile_ast_node_to_ir(schema, current_schema_type, ast, location, context)
     return basic_blocks
 
 
+def _validate_all_tags_are_used(context):
+    tag_names = set([tag_name for tag_name in context['tags']])
+    filter_args = set([])
+    for location, _ in context['metadata'].registered_locations:
```

Comment:
I'd kind of rather do this in the opposite direction -- let's make sure the metadata table carries enough information to determine which tags are defined, and which are used. That will make the code a lot more stable since it won't be as deeply coupled to the various parts of the `context` in the frontend.

Reply:
Added tag info into metadata in new commits. Do you have any thoughts on how this is implemented before I go and add more tests to `test_explain_info.py`?